packages/ibm-products/src/components/ConditionBuilder/ConditionGroupBuilder/ConditionGroupBuilder.js

```diff
@@ -45,12 +46,46 @@ const ConditionGroupBuilder = ({
       variant == 'tree'
         ? handleFocusOnCloseTree(evt)
         : handleFocusOnClose(evt, conditionIndex);
-      onChange({
-        ...group,
-        conditions: group.conditions.filter(
-          (condition) => conditionId !== condition.id
-        ),
-      });
+
+      if (group.conditions[1].conditions) {
+        //when we remove every plain conditions of a group without deleting  the subgroup, we need to restructure the group
+        //the inner group become outer group and same level subgroups become plain conditions
+
+        let newGroup = [],
+          allConditions = [];
+        group.conditions.map((eachItem) => {
+          if (eachItem.conditions) {
+            allConditions = [...allConditions, ...eachItem.conditions];
+          } //spreading out the condition inside the subgroup
+        });
+        //we always have conditions first and then subgroups, so ordering accordingly
+        const groupedItems = allConditions.reduce((acc, item) => {
```

Comment:
we can eliminate some redundant code by modifying the checks
```
const groupedItems = allConditions.reduce((acc, item) => {
  if (item.conditions) {
    if (!acc.groups) {
      acc.groups = [];
    }
    acc.groups.push(item);
  } else {
    if (!acc.conditions) {
      acc.conditions = [];
    }
    acc.conditions.push(item);
  }
  return acc;
}, {});
```

Reply:
Did some overall refactor in the logic.